kubectl-cloudflow/cmd/deploy.go

```diff
@@ -81,32 +105,198 @@ the stored credentials.
+				fmt.Printf("Deploying blueprint %s\n", cmd.Flag("blueprint").Value.String())
+			}
+		},
+		Run: deployOpts.deployImpl,
 	}
+	deployOpts.cmd.Flags().StringVarP(&deployOpts.blueprintFilePath, "blueprint", "b", "", "blueprint file path.")
+	deployOpts.cmd.Flags().StringVarP(&deployOpts.imagePath, "image", "i", "", "docker image path.")
 	deployOpts.cmd.Flags().StringVarP(&deployOpts.username, "username", "u", "", "docker registry username.")
 	deployOpts.cmd.Flags().StringVarP(&deployOpts.password, "password", "p", "", "docker registry password.")
 	deployOpts.cmd.Flags().BoolVarP(&deployOpts.passwordStdin, "password-stdin", "", false, "Take the password from stdin")
 	deployOpts.cmd.Flags().StringArrayVar(&deployOpts.volumeMounts, "volume-mount", []string{}, "Accepts a key/value pair separated by an equal sign. The key should be the name of the volume mount, specified as '[streamlet-name].[volume-mount-name]'. The value should be the name of an existing persistent volume claim.")
+	deployOpts.cmd.Flags().StringToIntVar(&deployOpts.replicas, "scale", map[string]int{}, "Accepts key/value pairs for replicas per streamlet")
 
 	rootCmd.AddCommand(deployOpts.cmd)
 }
 
 func (opts *deployOptions) deployImpl(cmd *cobra.Command, args []string) {
 	version.FailOnProtocolVersionMismatch()
 
-	imageRef := args[0]
+	if (cmd.Flag("image").Changed == true) && (cmd.Flag("blueprint").Changed == true) {
+		util.LogAndExit("%s", "Cannot specify both blueprint and image in command")
+	}
+	if (cmd.Flag("image").Changed == false) && (cmd.Flag("blueprint").Changed == false) {
+		util.LogAndExit("%s", "Need to specify either blueprint or image in command")
+	}
+	// if (cmd.Flag("image").Changed == true) && (cmd.Flag("scale").Changed == true) {
+	// util.LogAndExit("%s", "scale is supported only with blueprint based deployment")
+	// }
+	if cmd.Flag("image").Changed == true {
+		deployImage(cmd, opts, args)
+	} else {
+		deployBlueprint(cmd, opts, args)
+	}
+}
+
+func deployBlueprint(cmd *cobra.Command, opts *deployOptions, args []string) {
+	// read blueprint file and load its content
+	blueprintFile := cmd.Flag("blueprint").Value.String()
+	content, err := ioutil.ReadFile(blueprintFile)
+	if err != nil {
+		util.LogAndExit("%s", err.Error())
+	}
+	config, err := loadBlueprintConfigContent(string(content[:]))
+	if err != nil {
+		util.LogAndExit("%s", err.Error())
+	}
+	fmt.Printf("%s\n", config)
+
+	// get the images from the blueprint
+	images := config.GetNode("blueprint.images").GetObject().Items()
+
+	var imageRefs []deploy.ImageReference
+	var dockerRegistryURL string
+
+	for _, imageRef := range images {
+		ref, err := parseImageReference(imageRef.GetString())
+		if err != nil {
+			util.LogAndExit("%s", err.Error())
+		}
+		imageRefs = append(imageRefs, *ref)
+		dockerRegistryURL = ref.Registry
+	}
+
+	applicationSpec, pulledImages := deploy.GetCloudflowApplicationSpecFromDockerImages(imageRefs, config, opts.replicas)
```

Comment:
The name of this method is a bit confusing, and I guess the config passed in here is the blueprint?

Reply:
I guess this creates an cloudflow application CR from the blueprint and the images

Reply:
pulledImages, do we need to pull them all? or does it inspect?

Reply:
* Yes, the function creates an application CR from images and blueprint. Will `GetCloudflowApplicationSpecFromDockerImagesAndBlueprint` be a better name ? I tried to make it a bit terser :-)

* Yes, the `config` refers to that of the blueprint. I can change the name of the variable to make that explicit.

Reply:
* regarding `pulledImages` I will check if I can optimize it. It's basically the same stuff that was done before with images.